contracts/src/peer_protocol.cairo

```diff
@@ -385,6 +432,108 @@ pub struct BorrowProposalCreated {
             self.user_transactions.entry((user, current_count)).write(transaction);
             self.user_transactions_count.entry(user).write(current_count + 1);
         }
+
+        fn zero_address(self: @ContractState) -> ContractAddress {
+            contract_address_const::<0>()
+        }
+
+        fn handle_borrower_acceptance(ref self: ContractState, proposal: Proposal, lender: ContractAddress, net_amount: u256, fee_amount: u256) {
+            // Check if acceptor (lender) has sufficient funds
+            let lender_balance = self.token_deposits.entry((lender, proposal.token)).read();
+            assert(lender_balance >= proposal.amount, 'insufficient lender balance');
+
+            // Transfer net amount to borrower
+            IERC20Dispatcher { contract_address: proposal.token }.transfer(proposal.borrower, net_amount);
+
+            // Transfer protocol fee to protocol fee address
+            IERC20Dispatcher { contract_address: proposal.token }.transfer(self.protocol_fee_address.read(), fee_amount);
+
+            // Mint SPOK
+            self.mint_spoks(proposal.borrower, lender);
```

Comment:
<img width="1512" alt="Screenshot 2024-11-24 at 05 49 44" src="https://github.com/user-attachments/assets/6d5fcd3b-b066-431f-a1e7-366772a942bc">

Reply:
@Dprof-in-tech I just saw the issue. `src` was added to `gitignore` making newly created files hidden.